src/pages/index.js

```diff
@@ -12,54 +11,140 @@ import {
+    });
+    user.changeAvatarImage(userData.avatar);
+    section.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error(`Failed to load app info: ${err}`);
+  });
 
 //instantiate
 const section = new Section(
   {
-    items: initialCards,
     renderer: (data) => {
       section.addItem(createCard(data));
     },
   },
   cardSelectors.cardListEl,
 );
 
-const newCardModal = new ModalWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit,
-);
+const newCardModal = new ModalWithForm({
+  modalSelector: "#add-card-modal",
+  handleFormSubmit: (data) => {
+    newCardModal.renderLoading(true);
+    api
+      .addCard({
+        name: data.title,
+        link: data.link,
+      })
+      .then((data) => {
+        section.addItem(createCard(data));
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        newCardModal.renderLoading(false);
+        newCardModal.close();
+      });
+  },
+});
 
-const editProfileModal = new ModalWithForm(
-  "#profile-edit-modal",
-  handleProfileEditFormSubmit,
-);
+const editProfileModal = new ModalWithForm({
+  modalSelector: "#profile-edit-modal",
+  handleFormSubmit: (data) => {
+    editProfileModal.renderLoading(true);
+    api
+      .editProfile({
+        name: data.title,
+        about: data.bio,
+      })
+      .then((data) => {
+        user.setUserInfo({ name: data.name, bio: data.about });
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        editProfileModal.renderLoading(false);
+        editProfileModal.close();
+      });
+  },
+});
+
+const confirmDeleteModal = new ModalWithForm({
+  modalSelector: "#confirm-delete-modal",
+  handleFormSubmit: confirmDeleteCard,
+});
+
+const changeAvatarModal = new ModalWithForm({
+  modalSelector: "#change-avatar-modal",
+  handleFormSubmit: (data) => {
+    changeAvatarModal.renderLoading(true);
+    api
+      .changeAvatar(data.link)
+      .then((userData) => {
+        user.changeAvatarImage(userData.avatar);
+        changeAvatarModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        changeAvatarModal.renderLoading(false);
+        changeAvatarModal.close();
+      });
+  },
+});
 
 const imageModal = new ModalWithImage(cardSelectors.previewModal);
 
 const user = new UserInfo({
   name: ".profile__title",
   bio: ".profile__bio",
+  avatar: ".profile__avatar",
 });
 
 // //initialize instances
-section.renderItems(initialCards);
+
 imageModal.setEventListeners();
 newCardModal.setEventListeners();
 editProfileModal.setEventListeners();
+confirmDeleteModal.setEventListeners();
+changeAvatarModal.setEventListeners();
 
 // /* Validation */
 
 const editFormValidator = new FormValidator(settings, profileEditForm);
 const addFormValidator = new FormValidator(settings, addCardForm);
+const changeAvatarFormValidator = new FormValidator(settings, changeAvatarForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```